experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -163,11 +165,11 @@ function composeMarks<TNodeChange>(
 	revisionMetadata: RevisionMetadataSource,
 ): Mark<TNodeChange> {
 	const nodeChange = composeChildChanges(baseMark.changes, newMark.changes, newRev, composeChild);
-
-	if (isTransientEffect(newMark)) {
-		const newDetachRevision = newMark.detach.revision ?? newRev;
+	if (isTransientEffect(newMark) || isDetachOfRemovedNodes(newMark)) {
```

Comment:
Maybe we could name the type `TransientEffect | DetachOfRemovedNodes` something like HasTransientEffect. Maybe we could rename transient marks to something else and use the term transient for both of these. It seems like this line is checking "is this equivalent to a transient effect?" and `asTransient` takes "something that can be converted to a transient effect." It would be nice to factor the code to better reflect that.

Reply:
I also though about the name `HasTransientEffect` but dimissed it on the grounds that it would sounds too much like the existing `TransientEffect`. I clearly should have dreamt a little bigger an considered "what if we rename `TransientEffect` to something else?". Maybe `TransientEffect` could be renamed to `AttachAndDetach`.

Reply:
`AttachAndDetach` is at least very clear.

Now that I think about it, saying that these marks (an attach and detach, or a detach of removed nodes) have a transient effect seems a bit cryptic and perhaps misleading. Wouldn't it be more accurate to say that they rename an empty cell? So we could use try a name along the lines of EmptyCellRename.

Reply:
That's a good point, but I find EmptyCellRename a bit awkward because those marks do currently convey more than that in some cases (new inserts convey the raw data to build, and move-in+move-out also convey stuff about the move chain). All in all though, I'd rather have a slightly ill fitting name that more in tune with what we're trying to do, so EmptyCellRename (or just CellRename) would be my top choice. This would match the pattern we already have of Noop marks having nested changes (which also is a bit awkward).

Reply:
Since CellRename would just be a helper type, I wouldn't infer that that's the only effect of the mark.